lib/elixir/src/elixir_expand.erl

```diff
@@ -921,6 +922,26 @@ assert_generator_start(Meta, _, E) ->
 
 %% Assertions
 
+refute_parallel_bitstring_match({'<<>>', _, _}, {'<<>>', Meta, _} = Arg, E, true) ->
+  form_error(Meta, ?key(E, file), ?MODULE, {parallel_bitstring_match, Arg});
+refute_parallel_bitstring_match(Expr, {'=', _Meta, [Left, Right]}, E, Parallel) ->
+  refute_parallel_bitstring_match(Expr, Left, E, true),
+  refute_parallel_bitstring_match(Expr, Right, E, Parallel);
+refute_parallel_bitstring_match(Expr, Arg, E, Parallel) when is_list(Expr), is_list(Arg) ->
```

Comment:
We can also pattern match here:

    refute_parallel_bitstring_match([_ | _] = Expr, [_ | _] = Arg, E, Parallel) ->

Especially because we don't care about empty lists.